yb-voyager/src/queryparser/query_parser.go

```diff
@@ -163,4 +163,74 @@ func getCreateViewNode(parseTree *pg_query.ParseResult) (*pg_query.Node_ViewStmt
 func getCreateFuncStmtNode(parseTree *pg_query.ParseResult) (*pg_query.Node_CreateFunctionStmt, bool) {
 	node, ok := parseTree.Stmts[0].Stmt.Node.(*pg_query.Node_CreateFunctionStmt)
 	return node, ok
-}
\ No newline at end of file
+}
+
+func IsFunctionObject(parseTree *pg_query.ParseResult) bool {
+	funcNode, ok := getCreateFuncStmtNode(parseTree)
+	if !ok {
+		return false
+	}
+	return !funcNode.CreateFunctionStmt.IsProcedure
+}
+
+/*
+return type ex-
+CREATE OR REPLACE FUNCTION public.process_combined_tbl(
+    ... 
+)
+RETURNS public.combined_tbl.maddr%TYPE AS  
+return_type:{names:{string:{sval:"public"}}  names:{string:{sval:"combined_tbl"}}  names:{string:{sval:"maddr"}} 
+pct_type:true  typemod:-1  location:226}
+*/
+func GetReturnTypeOfFunc(parseTree *pg_query.ParseResult) string {
+	funcNode, _ := getCreateFuncStmtNode(parseTree)
+	returnType := funcNode.CreateFunctionStmt.GetReturnType()
+	return getParserTypeInString(returnType)
+}
+
+func getQualifiedTypeName(typeNames []*pg_query.Node) string {
+	var typeNameStrings []string
+	for _, n := range typeNames {
+		typeNameStrings = append(typeNameStrings, n.GetString_().Sval)
+	}
+	return strings.Join(typeNameStrings, ".")
+}
+
+func getParserTypeInString(typeVar *pg_query.TypeName) string {
```

Comment:
What do you mean by `ParserType` ?  isn't this function more about getting fully qualified type name?

Reply:
It's more of getting a string representation for fully qualified type name with `%TYPE` suffix from parser `TypeName`